actions/docker-build/action.yaml

```diff
@@ -0,0 +1,44 @@
+name: "Publish Docker image"
+description: "Use a Dockerfile to build an `image.tar` file and upload it to GitHub"
+
+inputs:
+  dockerfile-dir:
+    description: "Directory containing the Dockerfile"
+    required: true
+  github-token:
+    description: "GitHub token"
+    required: true
+  image-artifact-name:
+    description: "Name of the artifact that contains the Docker image.tar file to push, see https://github.com/actions/upload-artifact (Default is 'image-artifact')."
+    required: false
+    default: "image-artifact"
+  image-name:
+    description: "Name of Docker image (Default is the repository name)."
+    required: false
+    default: ${{ github.event.repository.name }}
+  ref:
+    description: "Ref to checkout"
+    required: false
+    default: ${{ github.ref_name }}
+runs:
+  using: "composite"
+  steps:
+    - name: Check out repository
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ inputs.ref }}
+        token: ${{ inputs.github-token }}
```

Comment:
Why do you need a token here to checkout the repository? We don't need it in other actions or workflows

Reply:
But this action is not creating the tag right? I don't mind optionally allowing the user to set a `github-token`, but it shouldn't be required then since this action can also be used directly and on other branches